Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Globalization;
+
+namespace BootCamp.Chapter
+{
+    public static class Lesson4
+    {
+        /// <summary>
+        /// Indication that the user input for an integer was empty.
+        /// </summary>
+        public const int EmptyNumberInput = 0;
+
+        /// <summary>
+        /// Indication that the user input for an integer was not in range.
+        /// </summary>
+        public const int InvalidNumberInput = -1;
+
+        /// <summary>
+        /// Indication that the user input for a name string was empty.
+        /// </summary>
+        public const string EmptyName = "-";
+
+        /// <summary>
+        /// This function will prompt some user data like name and age and in the end it will print out
+        /// the users BMI.
+        /// </summary>
+        public static void Demo()
+        {
+            ProcessPerson();
+            ProcessPerson();
+        }
+
+        /// <summary>
+        /// Prints the message and requests for user input of an integer.
+        /// </summary>
+        /// <param name="message">The message to print.</param>
+        /// <returns>The parsed integer if the input was valid, 0 when empty and -1 when invalid.</returns>
+        public static int PromptInt(string message)
+        {
+            return !PromptTryParse(message, out int parsingResult, out string input)
+                ? InvalidInput(input)
+                : RangeCheck(parsingResult);
+        }
+
+        /// <summary>
+        /// Prints the message and requests for user input of a float.
+        /// </summary>
+        /// <param name="message">The message to print.</param>
+        /// <returns>The parsed float if the input was valid, 0 when empty and -1 when invalid.
+        public static float PromptFloat(string message)
+        {
+            return !PromptTryParse(message, out float parsingResult, out string input)
+                ? InvalidInput(input)
+                : RangeCheck(parsingResult);
+        }
+
+        /// <summary>
+        /// Request for user name input.
+        /// </summary>
+        /// <param name="message">The request message.</param>
+        /// <returns>The user input.</returns>
+        public static string PromptString(string message)
+        {
+            return IsValidName(PromptInput(message));
```

Comment:
I would expect IsValidName to be true or false, yet here I see it return a string

Reply:
Sadly that is true. Any suggestions how to name the function?
Everything I come up with feels bad.

Reply:
IsValidName should be CorrectName or FixName

Reply:
Correct name also feels bad to me.
FixName it is!